### PR TITLE
feat(cleanup): add support for tenant reset/cleanup

### PR DESCRIPTION
**Pull request by dgutride**
Add a new cleanup component (linked to from the user profile page).  This shows all spaces, deletes them after sufficient warning for the user and calls the new tenant cleanup API (which deletes all openshift objects).  

Upon success, the tenant reset API is called which adds back Jenkins and Che support.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@dgutride fabric8/fabric8-ui:SNAPSHOT-PR-1897-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1897-fabric8-ui.badger.fabric8.io

File: src/app/profile/cleanup/cleanup.component.html [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                name="username"
+                [(ngModel)]="userName" />
+            </div>
+          </div>
+        </form>
+      </div>
+    </section>
+    <footer>
+      <div class="col-md-offset-4 col-md-8">
+        <button class="btn" [ngClass]="{ 'btn-danger': userNameMatches() }" type="button" (click)="confirm()" [disabled]="!userNameMatches()" type="submit"> I understand my actions - erase my environment</button>
```
Comment (joshuawilson) [inline src/app/profile/cleanup/cleanup.component.html]: You have 2 "type" in this line. A button and a submit.

File: src/app/profile/services/tenent.service.ts [hunk 2]
```diff
@@ -42,6 +42,22 @@ export class TenentService {
       });
   }
 
+  /**
+   * Cleanup tenant
+   * @returns {Observable<T>}
```
Comment (joshuawilson) [inline src/app/profile/services/tenent.service.ts]: Should the "T" be "any", just to keep the docs clear.

**Comment by joshuawilson** [comment]
We are going to need to clean out the spaces that are deleted from the context menu and the list of work items that were in the space that was deleted.
I'm not allowed to click on them but they are there. 
However they go away with a hard refresh.

**Pull request merged**